scalding-core/src/main/scala/com/twitter/scalding/typed/Sketched.scala

```diff
@@ -121,6 +129,17 @@ case class SketchJoined[K: Ordering, V, V2, R](left: Sketched[K, V],
       .withReducers(numReducers)
       .map{ case ((r, k), v) => (k, v) }
   }
+
+  private implicit def intKeyOrd: Ordering[(Int, K)] = {
+    val kord = implicitly[Ordering[K]]
+
+    kord match {
+      case kos: OrderedSerialization[_] => new OrderedSerialization2(ordSer[Int], kos.asInstanceOf[OrderedSerialization[K]])
+      case _ => Ordering.Tuple2[Int, K]
+    }
+  }
+
+  private implicit def ordSer[T]: OrderedSerialization[T] = macro OrderedSerializationProviderImpl[T]
```

Comment:
I am calling it in intKeyOrd method



    _____________________________
From: ianoc <notifications@github.com>
Sent: Tuesday, June 9, 2015 6:23 PM
Subject: Re: [scalding] Make SketchJoin ordered serialization aware (#1316)
To: twitter/scalding <scalding@noreply.github.com>
Cc: Mansur Ashraf <mansoor.ashraf@gmail.com>


    

In scalding-core/src/main/scala/com/twitter/scalding/typed/Sketched.scala:   > @@ -121,6 +129,17 @@ case class SketchJoined[K: Ordering, V, V2, R](left: Sketched[K, V],>        .withReducers(numReducers)>        .map{ case ((r, k), v) => (k, v) }>    }> +> +  private implicit def intKeyOrd: Ordering[(Int, K)] = {> +    val kord = implicitly[Ordering[K]]> +> +    kord match {> +      case kos: OrderedSerialization[_] => new OrderedSerialization2(ordSer[Int], kos.asInstanceOf[OrderedSerialization[K]])> +      case _ => Ordering.Tuple2[Int, K]> +    }> +  }> +> +  private implicit def ordSer[T]: OrderedSerialization[T] = macro OrderedSerializationProviderImpl[T]   

You can put it in scalding-serialization-macros. You can't call it there, but you can define it there, we have it in most of our macro modules.   

—
Reply to this email directly or view it on GitHub.